From: Junjie Fu <fujunjie1@qq.com>
Subject: mempolicy.h: remove unnecessary header file inclusions
Date: Fri, 6 Dec 2024 23:53:49 +0800

Originally, linux/mempolicy.h included linux/pagemap.h because
vma_migratable() was implemented inline within the header, requiring
mapping_gfp_mask() function to implement vma_migratable().  Now that
vma_migratable() is only declared in linux/mempolicy.h and its
implementation has been moved to mempolicy.c, the inclusion of
linux/pagemap.h in the header is no longer necessary.

Additionally, since mempolicy.c includes internal.h, and internal.h
already includes linux/pagemap.h, so there is no need to modify
mempolicy.c after removing the direct inclusion of linux/pagemap.h from
linux/mempolicy.h

Link: https://lkml.kernel.org/r/tencent_08B979048FE091821B290B18AE97E70DC507@qq.com
Signed-off-by: Junjie Fu <fujunjie1@qq.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/mempolicy.h |    1 -
 1 file changed, 1 deletion(-)

--- a/include/linux/mempolicy.h~mempolicyh-remove-unnecessary-header-file-inclusions
+++ a/include/linux/mempolicy.h
@@ -12,7 +12,6 @@
 #include <linux/rbtree.h>
 #include <linux/spinlock.h>
 #include <linux/nodemask.h>
-#include <linux/pagemap.h>
 #include <uapi/linux/mempolicy.h>
 
 struct mm_struct;
_
